crypto/curve25519/curve25519.c

```diff
@@ -4639,6 +4638,7 @@ int ED25519_keypair(uint8_t out_public_key[32], uint8_t out_private_key[64],
 
   ge_p3 A;
   x25519_ge_scalarmult_base(&A, az);
+  uint8_t out_public_key[32];
   ge_p3_tobytes(out_public_key, &A);
```

Comment:
Please just do `ge_p3_tobytes(&out_private_key[32], &A)` and remove `out_public_key` and `memmove` below. In particular, `out_` is for output parameters, and `out_public_key` isn't an output parameter so it shouldn't be named like that.